models/repo.go

```diff
@@ -1665,6 +1692,125 @@ func GitGcRepos() error {
+const STAT_SIZE = "size: "
+const STAT_INPACK = "in-pack: "
+const STAT_PACKS = "packs: "
+const STAT_SIZEPACK = "size-pack: "
+const STAT_PRUNEPACKAGE = "prune-package: "
+const STAT_GARBAGE = "garbage: "
+const STAT_SIZEGARBAGE = "size-garbage: "
+
+func GitRepoSize(repoPath string) (*CountObject, error) {
+	var cmd *exec.Cmd
+	cmd = exec.Command("git", "count-objects", "-v")
+	cmd.Dir = repoPath
+	cmd.Stderr = os.Stderr
+
+	stdout, err := cmd.StdoutPipe()
+	if err != nil {
+		return nil, fmt.Errorf("StdoutPipe: %v", err)
+	}
+	if err = cmd.Start(); err != nil {
+		return nil, fmt.Errorf("Start: %v", err)
+	}
+
+	repoSize, err := ParseSize(stdout)
+	if err != nil {
+		return nil, fmt.Errorf("ParsePatch: %v", err)
+	}
+
+	if err = cmd.Wait(); err != nil {
+		return nil, fmt.Errorf("Wait: %v", err)
+	}
+
+	return repoSize, nil
+}
+
+func ParseSize(reader io.Reader) (*CountObject, error) {
+	repoSize := &CountObject{}
+	input := bufio.NewReader(reader)
+	isEOF := false
+
+	for !isEOF {
+		line, err := input.ReadString('\n')
+		if err != nil {
+			if err == io.EOF {
+				isEOF = true
+				continue
+			} else {
+				return nil, fmt.Errorf("ReadString: %v", err)
+			}
+		}
+
+		if len(line) > 0 && line[len(line)-1] == '\n' {
+			// Remove line break.
+			line = line[:len(line)-1]
+		}
+
+		switch {
+		case strings.HasPrefix(line, STAT_COUNT):
+			repoSize.Count, err = strconv.Atoi(line[7:])
+		case strings.HasPrefix(line, STAT_SIZE):
+			repoSize.Size, err = strconv.ParseInt(line[6:], 10, 64)
+			repoSize.Size = repoSize.Size * 1024
+		case strings.HasPrefix(line, STAT_INPACK):
+			repoSize.InPack, err = strconv.ParseInt(line[9:], 10, 64)
+			repoSize.InPack = repoSize.InPack * 1024
+		case strings.HasPrefix(line, STAT_PACKS):
+			repoSize.Packs, err = strconv.Atoi(line[7:])
+		case strings.HasPrefix(line, STAT_SIZEPACK):
+			repoSize.SizePack, err = strconv.ParseInt(line[11:], 10, 64)
+			repoSize.SizePack = repoSize.SizePack * 1024
+		case strings.HasPrefix(line, STAT_PRUNEPACKAGE):
+			repoSize.PrunePack, err = strconv.Atoi(line[16:])
+		case strings.HasPrefix(line, STAT_GARBAGE):
+			repoSize.Garbage, err = strconv.Atoi(line[9:])
+		case strings.HasPrefix(line, STAT_SIZEGARBAGE):
+			repoSize.SizeGarbage, err = strconv.Atoi(line[14:])
+		}
+		if err != nil {
+			log.Error(4, "Parsing count-objects failed: %v", err)
+			return nil, err
+		}
+	}
+	return repoSize, nil
+}
+
+func UpdateRepoSize(repoUserName, repoName string) error {
+	log.Info("Compute Size of %s - %s", repoUserName, repoName)
+
+	repoPath := RepoPath(repoUserName, repoName)
+	repoInfoSize, err := GitRepoSize(repoPath)
+	if err != nil {
+		return err
+	}
+	repoSize := repoInfoSize.Size + repoInfoSize.SizePack
+	log.Info(strconv.FormatInt(repoSize, 10))
+	sess := x.NewSession()
+	defer sessionRelease(sess)
+	if err = sess.Begin(); err != nil {
+		return nil
+	}
+	if _, err = sess.Exec("UPDATE `repository` SET repo_size=? WHERE name=?", repoSize, repoName); err != nil {
```

Comment:
I have done the same thing as previous SQL ([See L1372](https://github.com/go-gitea/gitea/blob/master/models/repo.go#L1372)).
I will do it in XORM but only mine.

A review of SQL queries need to be done.